packages/test/test-end-to-end-tests/src/test/summarizeIncrementally.spec.ts

```diff
@@ -0,0 +1,245 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { ISummaryTree, SummaryType } from "@fluidframework/protocol-definitions";
+import { IContainerRuntimeBase, channelsTreeName } from "@fluidframework/runtime-definitions";
+import {
+	ITestObjectProvider,
+	createSummarizer,
+	getContainerEntryPointBackCompat,
+	summarizeNow,
+	waitForContainerConnection,
+} from "@fluidframework/test-utils";
+import {
+	ITestDataObject,
+	TestDataObjectType,
+	describeCompat,
+} from "@fluid-private/test-version-utils";
+import { IContainer } from "@fluidframework/container-definitions";
+import { ISummarizer } from "@fluidframework/container-runtime";
+
+/**
+ * Validates that the data store summary is as expected.
+ * If expectHandle is false, the data store summary should be a tree.
+ * If expectHandle is true, the data store summary should be a handle and the handle id should be correct.
+ */
+function validateDataStoreStateInSummary(
+	summaryTree: ISummaryTree,
+	dataStoreId: string,
+	expectHandle: boolean,
```

Comment:
>expectHandle [](http://example.com/codeflow?start=1&length=12)

observation: I noticed we don't start boolean var with `f` prefix throughout our codebase. Improves readability where I've seen this being done in Word/Excel/PPT codebase.

Reply:
True. We don't follow these naming conventions for anything other than interfaces (which start with I).